lib/galaxy/security/validate_user_input.py

```diff
@@ -65,24 +67,27 @@ def validate_publicname(trans, publicname, user=None):
     return ''
 
 
-def transform_publicname(trans, publicname, user=None):
-    # User names must be at least four characters in length and contain only lower-case
-    # letters, numbers, and the '-' character.
+def transform_publicname(trans, publicname, user=None, check_dup=True):
+    """
+    Transform publicname to respect the minimum and maximum string length, and
+    the allowed characters.
+    FILL_CHAR is used to extend or replace characters.
+    """
     # TODO: Enhance to allow generation of semi-random publicnnames e.g., when valid but taken
     if user and user.username == publicname:
```

Comment:
Do you have any use case in mind where `transform_publicname` is called with an existing user and a matching publicname ?

Reply:
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/webapps/tool_shed/controllers/user.py#L157